soh/soh/Enhancements/randomizer/3drando/spoiler_log.cpp

```diff
@@ -308,25 +308,41 @@ static void WriteLocation(
 
 //Writes a shuffled entrance to the specified node
 static void WriteShuffledEntrance(
-  tinyxml2::XMLElement* parentNode,
-  Entrance* entrance,
-  const bool withPadding = false
-) {
-  auto node = parentNode->InsertNewChildElement("entrance");
-  node->SetAttribute("name", entrance->GetName().c_str());
-  auto text = entrance->GetConnectedRegion()->regionName + " from " + entrance->GetReplacement()->GetParentRegion()->regionName;
-  node->SetText(text.c_str());
-
-  if (withPadding) {
-    constexpr int16_t LONGEST_NAME = 56; //The longest name of a vanilla entrance
-
-    //Insert padding so we get a kind of table in the XML document
-    int16_t requiredPadding = LONGEST_NAME - entrance->GetName().length();
-    if (requiredPadding > 0) {
-      std::string padding(requiredPadding, ' ');
-      node->SetAttribute("_", padding.c_str());
+    Entrance* entrance, const bool withPadding = false) {
+    // Entrance* entrance = Entrances(EntranceKey);
+    int16_t originalIndex = entrance->GetIndex();
+    int16_t destinationIndex = entrance->GetReverse()->GetIndex();
+    int16_t originalBlueWarp = entrance->GetBlueWarp();
+    int16_t replacementIndex = entrance->GetReplacement()->GetIndex();
+    int16_t replacementDestinationIndex = entrance->GetReplacement()->GetReverse()->GetIndex();
+  // auto node = parentNode->InsertNewChildElement("location");
+  switch (gSaveContext.language) {
+        case LANGUAGE_ENG:
+        case LANGUAGE_FRA:
+        default:
+            jsonData["Entrances"][std::to_string(originalIndex)] = replacementIndex;
+            jsonData["Entrances"][std::to_string(replacementDestinationIndex)] = destinationIndex;
```

Comment:
I know bria mentioned this already but just so it's in the review, we'll want these to be human readable names in the spoiler file. You'll probably need an `unordered_map` like the existing ones for check names and item names in `randomizer.cpp`. You'll probably also want an enum in `randomizerTypes.h` to give each number a descriptive name for the code to use.